### PR TITLE
feat: Authentication - refresh access token endpoint

### PR DESCRIPTION
**Pull request by King-AJr**
## What does this PR do?

This PR implements a new endpoint for refreshing access tokens using a refresh token in the NestJS application. It includes:

- **Added `POST /auth/token/refresh` Endpoint**: Allows clients to obtain a new access token using a valid refresh token.
- **Response Handling**: Provides appropriate responses for missing or invalid refresh tokens.
- **Error Handling**: Returns specific HTTP status codes and messages based on token validity.

## How should this be manually tested?

1. **Register a User**:
   - Use the `POST /auth/register` endpoint to register a new user.

2. **Generate Access and Refresh Tokens**:
   - After registration, ensure you obtain the access token and refresh token.

3. **Refresh Access Token**:
   - Make a `POST` request to `/auth/token/refresh` with the refresh token in the request body.
   - Verify that you receive a new access token if the refresh token is valid.
   - Check the response for appropriate status codes and messages if the refresh token is missing or invalid.

4. **Edge Cases**:
   - Test with an invalid or expired refresh token to ensure the endpoint returns a `401 Unauthorized` status with the correct message.
   - Test without including a refresh token in the request body to confirm a `400 Bad Request` status is returned.

## Checklist

- [x] Implemented `POST /auth/token/refresh` endpoint in the `auth.controller.ts`.
- [x] Added `refreshAccessToken` method in the `AuthenticationService`.
- [x] Updated `AuthenticationService` to handle refresh token validation and generation of new access tokens.
- [x] Modified `auth.controller.ts` to provide appropriate responses for various cases.
- [x] Updated sign up route to return refresh token on user sign up. 
- [x] Tested endpoint functionality to ensure it correctly handles valid and invalid tokens.
- [x] Added appropriate HTTP status codes and response messages for edge cases.
- [x] Added test for this endpoint

### CODE REVIEW

**Comment by King-AJr** [comment]
@TheCodeGhinux I need your review on this. Thanks

**Comment by King-AJr** [comment]
@Homoakin619 I need your review on this PR.

**Comment by davidolanrewaju** [comment]
Nicely done, but please include your documentations

File: src/modules/auth/auth.controller.ts
Comment (Homoakin619) [inline src/modules/auth/auth.controller.ts:1]: move the business logics here into a service such as the authservice, you should have just a line or two in your controller

**Comment by King-AJr** [comment]
@Cobidex @Homoakin619 I've added a secret for the refresh token and transferred the logic to the service

**Comment by King-AJr** [comment]
> Well done. Please resolve merge conflicts

I've resolved the conflicts

File: src/modules/auth/auth.service.ts [hunk 1]
```diff
@@ -74,4 +78,67 @@ export default class AuthenticationService {
...
+      },
+      {
+        secret: appConfig().jwtRefreshSecret,
+        expiresIn: appConfig().jwtRefreshExpiry,
+      }
+    );
+  }
+
+  async generateAccessToken(user: any) {
+    return this.jwtService.sign(
```
Comment (Cobidex) [inline src/modules/auth/auth.service.ts:98]: Well done on the implementation. the only issue i see is the storage of user personal information in the jwt token, its not best practice security wise. its better to have only the user id in it.

**Review by Cobidex — DISMISSED** [review]
In line with best practices it's better to store only the id in jwt tokens. Personal identifiers of the user such as email should not be in the token

**Comment by King-AJr** [comment]
@buka4rill greetings chief I need your review on this PR

File: package.json
Comment (Heba-WebDev) [inline package.json:1]: Please remove the double nodemailer dependencies

**Review by Heba-WebDev — CHANGES_REQUESTED** [review]
Please remove the nodemailer dependency and make sure you didn't do the same with passport (I see you installed it and I think it was installed before)

**Review by Heba-WebDev — APPROVED** [review]

**Review by buka4rill — COMMENTED** [review]
Looks ok but fix conflicts

**Comment by King-AJr** [comment]
@buka4rill I've dealt with the conflicts

**Comment by Heba-WebDev** [comment]
@King-AJr You still have conflicts (probably new) so please fix it

**Comment by King-AJr** [comment]
> @King-AJr You still have conflicts (probably new) so please fix it

@Heba-WebDev check again

**Comment by King-AJr** [comment]
@buka4rill I've dealt with the new merge conflicts

**Pull request closed without merging**